### ISSUE TITLE
1.45.3 breaks `self` descriptor

### ISSUE THREAD

**Issue by zloirock** [denoland/deno#24683]
`globalThis.self` should be a getter, but 1.45.3 breaks it - here it's a usual property:

<img width="344" alt="Снимок экрана 2024-07-23 в 09 11 13" src="https://github.com/user-attachments/assets/6bdb66f0-8bf0-4b8e-aad3-eb2b49e8117d">

Before that:

<img width="338" alt="Снимок экрана 2024-07-23 в 09 45 01" src="https://github.com/user-attachments/assets/49c47cb5-b0e7-4b29-a6f7-71d31e1c08ba">

That enforces `core-js` to load a polyfill and patch it.

**Comment by nathanwhit** [comment]
Looks like this regressed in 76b8ecbb6d8c07d29c34fb0b301cc3bf3351e3aa (cc @kt3k).

**Comment by kt3k** [comment]
Looks like the global var segregator (interceptor) in `ext/node/global.rs` doesn't transfer(?) property descriptor correctly from inner object at `Object.getOwnPropertyDescriptor` call.

The same issue seems happening to `window`, which is defined as getter-only property in `runtime/js/98_global_scope_window.js`, but `Object.getOwnPropertyDescriptor(globalThis, "window")` returns:

```js
{
  value: Window {},
  writable: true,
  enumerable: true,
  configurable: true
}
```

@lucacasonato Do you have any clue on this?

**Comment by kt3k** [comment]
Some notes:
- When `Object.getOwnPropertyDescriptor(globalThis, "self")` is called, `descriptor` callback (in `ext/node/global.rs`) is not called. Instead `getter` callback is called.
- When `ext_node_denoGlobals` is exposed to global, `Object.getOwnPropertyDescriptor(ext_node_denoGlobals, "self")` returns getter/setter descriptor. `definer` interceptor seems correctly working.

It looks strange that `getter` callback is called at `Object.getOwnPropertyDescriptor` call, but one solution (workaround) might be returning `v8::Intercepted::No` in `getter` intercepter for key `self` (and `window`).

**Comment by devsnek** [comment]
This is a bug in V8, caused by assuming that an object can only have one `ACCESS_CHECK`, and therefore never reaching the `INTERCEPTOR` state for the global object: https://source.chromium.org/chromium/chromium/src/+/main:v8/src/objects/js-objects.cc;l=1834;drc=2017cd8a8925f180257662f78eaf9eb93e8e394d?q=js-objects.cc&ss=chromium
I have submitted a patch here: https://chromium-review.googlesource.com/c/v8/v8/+/5737946

**Comment by zloirock** [comment]
It seems it's fixed in 1.46.0.
